proposals/mutate_existing_resources.md

```diff
@@ -0,0 +1,216 @@
+# Kyverno Design Proposal - Mutate Existing Resources
+
+Created: March 21st, 2022
+Author: [Shuting Zhao](https://github.com/realshuting)
+
+
+Table of contents:
+  * [Overview](#overview)
+  * [Proposal](#proposal)
+  * [Examples](#examples)
+  * [Implementation](#implementation)
+    + [Watch Trigger Resources](#watch-trigger-resources)
+      - [1. Admission Webhook](#1-admission-webhook)
+      - [2. Event Watcher](#2-event-watcher)
+    + [Apply Policies](#apply-policies)
+    + [Mutate Existing Resources](#mutate-existing-resources)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+
+## Overview
+
+Kyverno webhook controller runs as an admission controller in Kubernetes, and it mutates resources upon admission review. Such mutation is performed based on resources operations (CREATE/UPDATE/DELETE) via webhook.
+
+As of release 1.6.1, these are still the outstanding issues which Kyverno needs to support hence this proposal:
+
+- Mutate target resource which is different from the trigger resource, [#2139](https://github.com/kyverno/kyverno/issues/2139), [#1722](https://github.com/kyverno/kyverno/issues/1722).
+- Mutate trigger resources based on policy update, [#1607](https://github.com/kyverno/kyverno/issues/1607).
```

Comment:
May want to add https://github.com/kyverno/kyverno/issues/1722 since you call this out in your first message.

Reply:
Sorry I didn't get this, where do we want to add 1722?

Reply:
Yes, it's added here:
>- Mutate target resource which is different from the trigger resource, [#2139](https://github.com/kyverno/kyverno/issues/2139), [#1722](https://github.com/kyverno/kyverno/issues/1722).

Is this what you meant?

Reply:
Yes I think so.